lib/rules/no-misleading-character-class.js

```diff
@@ -197,32 +233,39 @@ module.exports = {
 
         /**
          * Generates a granular loc for context.report.
-         * @param {Character} char Individual character being reported on.
-         * @param {string} kind What kind of character report is being made.
+         * @param {Character[]} chars Individual characters being reported on.
          * @param {Node} node Parent string node to report within.
          * @returns {Object} Granular loc for context.report.
-         * @see https://github.com/eslint/eslint/pull/17515#issuecomment-1707697186
+         * @see https://github.com/eslint/eslint/pull/17515
          */
-        function generateReportLocation(char, kind, node) {
-            if (node.type === "TemplateLiteral" && node.expressions.length || sourceCode.getText(node).includes("\\")) {
+        function generateReportLocation(chars, node) {
```

Comment:
There's a lot of calculating going on in this function, can you add some comments to explain what's happening throughout?